File: common/src/main/java/com/pluxity/permission/dto/PermissionGroupCreateRequest.java
package com.pluxity.permission.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import java.util.List;

public record PermissionGroupCreateRequest(
        @NotNull @Schema(description = "권한 집합 이름") String name,
        @Schema(description = "권한에 대한 설명") String description,
        @NotNull @Schema(description = "권한의 상세 목록") List<PermissionRequest> permissions) {}

File: common/src/main/kotlin/com/pluxity/permission/dto/PermissionGroupCreateRequest.kt
package com.pluxity.permission.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull

data class PermissionGroupCreateRequest(
    @field:Schema(description = "권한 집합 이름")
    @field:NotNull
    val name: String,
    @field:Schema(description = "권한에 대한 설명")
    val description: String?,
    @field:Schema(description = "권한의 상세 목록")
    @field:NotNull
    val permissions: List<PermissionRequest>,
)
